patches/0083-lavfi-dnn-Add-OpenVINO-API-2.0-support.patch

```diff
@@ -0,0 +1,1224 @@
++        }
++        return 0;
++    } else {
++        ret = fill_model_input_ov2(ov2_model, request);
++        if (ret != 0) {
++            goto err;
++        }
++        status = ov_infer_request_infer(request->infer_request);
++        if (status != OK) {
++            av_log(NULL, AV_LOG_ERROR, "Failed to start synchronous model inference for OV2\n");
++            ret = DNN_GENERIC_ERROR;
++            goto err;
++        }
++        infer_completion_callback(request);
++        return (task->inference_done == task->inference_todo) ? 0 : DNN_GENERIC_ERROR;
++    }
++
++err:
++    return ret;
++}
++
++static int get_input_ov2(void *model, DNNData *input, const char *input_name)
++{
++    OV2Model *ov2_model = model;
++    OV2Context *ctx = &ov2_model->ctx;
++    ov_shape_t input_shape = {0};
++    ov_element_type_e tensor_type;
++    int64_t* dims;
++    ov_status_e status;
++    int input_resizable = ctx->options.input_resizable;
++
++    if (!ov_model_is_dynamic(ov2_model->ov_model)) {
++        status = ov_model_const_input_by_name(ov2_model->ov_model, input_name, &ov2_model->input_port);
++        if (status != OK) {
++            av_log(ctx, AV_LOG_ERROR, "Failed to get input port shape.\n");
++            return DNN_GENERIC_ERROR;
++        }
++
++        status = ov_const_port_get_shape(ov2_model->input_port, &input_shape);
++        if (status != OK) {
++            av_log(ctx, AV_LOG_ERROR, "Failed to get input port shape.\n");
++            return DNN_GENERIC_ERROR;
++        }
++        dims = input_shape.dims;
++
++        status = ov_port_get_element_type(ov2_model->input_port, &tensor_type);
++        if (status != OK) {
++            av_log(ctx, AV_LOG_ERROR, "Failed to get input port data type.\n");
++            return DNN_GENERIC_ERROR;
++        }
++    } else {
++        avpriv_report_missing_feature(ctx, "Do not support dynamic model now.");
++        return DNN_GENERIC_ERROR;
++    }
++
++    input->channels = dims[1];
++    input->height   = input_resizable ? -1 : dims[2];
++    input->width    = input_resizable ? -1 : dims[3];
++    input->dt       = port_datatype_to_dnn_datatype(tensor_type);
++
++    return 0;
++}
++
++static int contain_valid_detection_bbox(AVFrame *frame)
++{
++    AVFrameSideData *sd;
++    const AVDetectionBBoxHeader *header;
++    const AVDetectionBBox *bbox;
++
++    sd = av_frame_get_side_data(frame, AV_FRAME_DATA_DETECTION_BBOXES);
++    if (!sd) { // this frame has nothing detected
++        return 0;
++    }
++
++    if (!sd->size) {
++        return 0;
++    }
++
++    header = (const AVDetectionBBoxHeader *)sd->data;
++    if (!header->nb_bboxes) {
++        return 0;
++    }
++
++    for (uint32_t i = 0; i < header->nb_bboxes; i++) {
++        bbox = av_get_detection_bbox(header, i);
++        if (bbox->x < 0 || bbox->w < 0 || bbox->x + bbox->w >= frame->width) {
++            return 0;
++        }
++        if (bbox->y < 0 || bbox->h < 0 || bbox->y + bbox->h >= frame->width) {
```

Comment:
typo ? I guess it should be frame->height.

Reply:
can bbox->w or bbox->h be 0 ?

Reply:
Yes, it should be `frame->height` and `bbox->w bbox->h` cannot be 0.